acpi/src/fadt.rs

```diff
@@ -128,3 +131,27 @@ impl Fadt {
         }
     }
 }
+
+/// Information about the ACPI Power Management Timer (ACPI PM Timer).
+pub struct PmTimer {
+    /// An I/O space address to the register block of ACPI PM Timer.
+    pub io_base: u32,
```

Comment:
From §4.8.3.3 of the spec, it looks like the PM timer doesn't have to be in I/O space, but can instead be in system memory. `PM_TMR_BLK` should always be treated as an I/O address, but is superceded by the `X_PM_TMR_BLK` field (if present and non-zero), which is a [`GenericAddress`](https://github.com/rust-osdev/acpi/blob/master/acpi/src/lib.rs#L307). This should therefore probably be a `GenericAddress` (which should be created from `PM_TMR_BLK` in I/O space if that field is used.

Reply:
Do you mean like this?
```rust
enum Addr{
    Phys(usize),
    Io(u32),
    Pci(u32),
}
```

Reply:
Sort of, unfortunately we need to handle the rest of the address spaces off the bat as well as we'd break semver by adding the other variants later.

I've had a more careful look at how we should do this, and it turns out an enum doesn't really work as well as I thought it would. I basically ended up writing what I thought would work; hope that's okay. If you rebase your branch you should be able to switch to just using the new [`GenericAddress`](https://github.com/rust-osdev/acpi/blob/master/acpi/src/platform/address.rs#L53-L97).